src/mjolnir/directededgebuilder.cc

```diff
@@ -134,14 +134,24 @@ DirectedEdgeBuilder::DirectedEdgeBuilder(const OSMWay& way,
   if ((way.taxi_forward() && !forward) || (way.taxi_backward() && forward)) {
     reverse_access |= kTaxiAccess;
   }
-  if (way.pedestrian()) {
+  if ((way.pedestrian_forward() && forward) || (way.pedestrian_backward() && !forward)) {
     forward_access |= kPedestrianAccess;
+  }
+  if ((way.pedestrian_forward() && !forward) || (way.pedestrian_backward() && forward)) {
     reverse_access |= kPedestrianAccess;
   }
-  if (way.use() != Use::kSteps &&
-      ((way.wheelchair_tag() && way.wheelchair()) || (!way.wheelchair_tag() && way.pedestrian()))) {
-    forward_access |= kWheelchairAccess;
-    reverse_access |= kWheelchairAccess;
+  if (way.use() != Use::kSteps) {
+    if (way.wheelchair_tag() && way.wheelchair()) {
```

Comment:
Wheelchair routing and logic needs to be finished.  This was started a long time ago but never finished.